### PR TITLE
[CI environment] Old dependencies approach clean up

### LINKED ISSUE #2291: Old dependencies approach clean up
- Remove dependencies pipeline (GH + ADO)
- Remove ADO dependencies pipeline template
- Remove old dependencies pipeline references from the Wiki
- Remove specific code snippets supporting the old dependencies approach from actions and scripts

### PR DESCRIPTION
**Pull request by eriqua**
# Description

Closes #2291 

Depends on #1791 closure

- Remove dependencies pipeline (GH + ADO)
- Remove ADO dependencies pipeline template
- Remove old dependencies pipeline references from the Wiki
- Remove specific code snippets supporting the old dependencies approach from actions and scripts
- Remove Get-DependencyResourceNameList and references
- Update Wiki ci env deployment validation - Removal - how it works 
- Update Wiki getting started and fetch latest changes - deploy dependencies and update module test files

| Pipeline |
| - |
| [Compute: DiskEncryptionSets](https://github.com/Azure/ResourceModules/actions/workflows/ms.compute.diskencryptionsets.yml) |
| [Authorization: PolicyAssignments](https://github.com/Azure/ResourceModules/actions/workflows/ms.authorization.policyassignments.yml) |
| [Management: ManagementGroups](https://github.com/Azure/ResourceModules/actions/workflows/ms.management.managementgroups.yml) |
| [Build Status](https://dev.azure.com/carml/CARML/_build/latest?definitionId=30&branchName=users%2Ferikag%2Fdep-cleanup) |
| [Build Status](https://dev.azure.com/carml/CARML/_build/latest?definitionId=15&branchName=users%2Ferikag%2Fdep-cleanup) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

# Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

### CODE REVIEW

File: docs/wiki/Fetching latest changes - Scenario 2 Module library and CI environment.md [hunk 1]
```diff
@@ -11,9 +11,8 @@ The update process is the following:
 - [3. (Optional) Convert library to ARM](#3-optional-convert-library-to-arm)
 - [4. Apply specific settings to files](#4-apply-specific-settings-to-files)
 - [5. (Optional) Customize modules and CI environment](#5-optional-customize-modules-and-ci-environment)
-- [6. Update module test files](#6-update-module-test-files)
-- [7. Run dependencies pipeline](#7-run-dependencies-pipeline)
-- [8. Test and publish modules](#8-test-and-publish-modules)
+- [6. Run dependencies pipeline](#7-run-dependencies-pipeline)
```
Comment (AlexanderSehr) [inline docs/wiki/Fetching latest changes - Scenario 2 Module library and CI environment.md:14]: I guess, this should be removed?

Comment (AlexanderSehr) [reply docs/wiki/Fetching latest changes - Scenario 2 Module library and CI environment.md:14]: I think the wrong header was deleted

File: docs/wiki/Getting started - Scenario 2 Onboard module library and CI environment.md [hunk 2]
```diff
@@ -6,7 +6,6 @@ This requires several steps:
 1. [Fork/clone the repository into your DevOps environment](#2-forkclone-the-repository-into-your-devops-environment)
 1. [Configure the CI environment](#3-configure-the-ci-environment)
 1. [Deploy dependencies](#4-deploy-dependencies)
```
Comment (AlexanderSehr) [inline docs/wiki/Getting started - Scenario 2 Onboard module library and CI environment.md:8]: Should we explicity call it "manual" dependencies?

Comment (eriqua) [reply docs/wiki/Getting started - Scenario 2 Onboard module library and CI environment.md]: done

File: docs/wiki/Fetching latest changes - Scenario 2 Module library and CI environment.md (hunk 1 above)
Comment (eriqua) [reply docs/wiki/Fetching latest changes - Scenario 2 Module library and CI environment.md]: fixed

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**